src/drivers/roboclaw/RoboClaw.hpp

```diff
@@ -0,0 +1,298 @@
+	 */
+	RoboClaw(const char *deviceName, const char *baudRateParam);
+
+	/**
+	 * deconstructor
+	 */
+	virtual ~RoboClaw();
+
+	void initialize();
+
+	/**
+	 * @return position of a motor, rev
+	 */
+	float getMotorPosition(e_motor motor);
+
+	/**
+	 * @return speed of a motor, rev/sec
+	 */
+	float getMotorSpeed(e_motor motor);
+
+	/**
+	 * set the speed of a motor, rev/sec
+	 */
+	int setMotorSpeed(e_motor motor, float value);
+
+	/**
+	 * set the duty cycle of a motor
+	 */
+	int setMotorDutyCycle(e_motor motor, float value);
+
+	/**
+	 * Drive both motors. +1 = full forward, -1 = full backward
+	 */
+	int drive(float value);
+
+	/**
+	 * Turn. +1 = full right, -1 = full left
+	 */
+	int turn(float value);
+
+	/**
+	 * reset the encoders
+	 * @return status
+	 */
+	int resetEncoders();
+
+	/**
+	 * read data from serial
+	 */
+	int readEncoder();
+
+	/**
+	 * print status
+	 */
+	void printStatus();
+
+
+private:
+
+	char _storedDeviceName[256]; // Adjust size as necessary
+    	char _storedBaudRateParam[256]; // Adjust size as necessary
+
+	int _timeout_counter = 0;
+
+	// commands
+	// We just list the commands we want from the manual here.
+	enum e_command {
+
+		// simple
+		CMD_DRIVE_FWD_1 = 0,
+		CMD_DRIVE_REV_1 = 1,
+		CMD_DRIVE_FWD_2 = 4,
+		CMD_DRIVE_REV_2 = 5,
+
+		CMD_DRIVE_FWD_MIX = 8,
+		CMD_DRIVE_REV_MIX = 9,
+		CMD_TURN_RIGHT = 10,
+		CMD_TURN_LEFT = 11,
+
+		// encoder commands
+		CMD_READ_ENCODER_1 = 16,
+		CMD_READ_ENCODER_2 = 17,
+		CMD_READ_SPEED_1 = 18,
+		CMD_READ_SPEED_2 = 19,
+		CMD_RESET_ENCODERS = 20,
+		CMD_READ_BOTH_ENCODERS = 78,
+		CMD_READ_BOTH_SPEEDS = 79,
+
+		// advanced motor control
+		CMD_READ_SPEED_HIRES_1 = 30,
+		CMD_READ_SPEED_HIRES_2 = 31,
+		CMD_SIGNED_DUTYCYCLE_1 = 32,
+		CMD_SIGNED_DUTYCYCLE_2 = 33,
+
+		CMD_READ_STATUS = 90
+	};
+
+
+	struct {
+		speed_t serial_baud_rate;
+		int32_t counts_per_rev;
+		int32_t encoder_read_period_ms;
+		int32_t actuator_write_period_ms;
+		int32_t address;
+	} _parameters{};
+
+	struct {
+		param_t serial_baud_rate;
+		param_t counts_per_rev;
+		param_t encoder_read_period_ms;
+		param_t actuator_write_period_ms;
+		param_t address;
+	} _param_handles{};
+
+	int _uart;
+	fd_set _uart_set;
+	struct timeval _uart_timeout;
+
+	/** actuator controls subscription */
+	int _actuatorsSub{-1};
+	actuator_controls_s _actuatorControls;
+
+	int _armedSub{-1};
+	actuator_armed_s _actuatorArmed;
+
+	int _paramSub{-1};
+	parameter_update_s _paramUpdate;
+
+	uORB::Subscription _vehicle_thrust_setpoint_sub{ORB_ID(vehicle_thrust_setpoint)};
+	uORB::Subscription _vehicle_torque_setpoint_sub{ORB_ID(vehicle_torque_setpoint)};
+
+	vehicle_thrust_setpoint_s		vehicle_thrust_setpoint{};
+	vehicle_torque_setpoint_s		vehicle_torque_setpoint{};
+
+
+
+	uORB::PublicationMulti<wheel_encoders_s> _wheelEncodersAdv[2] { ORB_ID(wheel_encoders), ORB_ID(wheel_encoders)};
+	wheel_encoders_s _wheelEncoderMsg[2];
+
+	uint32_t _lastEncoderCount[2] {0, 0};
```

Comment:
Nitpick, we use snake case for field naming. 
```suggestion
	uint32_t _last_encoder_count[2] {0, 0};
```

Reply:
This one became obsolete during refactoring.